server/db/dbManager.ts

```diff
@@ -24,23 +24,104 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(parent?: mongoose.Types.ObjectId, depth: number = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
-		let posts;
-		if (parent) {
-			posts = await this.model.find({ parent }).sort({ upvotes: -1 }).lean().exec();
-			if (depth < parseInt(process.env.MAX_COMMENT_DEPTH)) {
-				for (const post of posts) {
-					post.children = [];
-					if (depth > 0) {
-						parentObject.children.push(post);
+	async getAllPosts(recursive: boolean, parent?: mongoose.Types.ObjectId, parentObject = { children: [] }): Promise<object> {
+		if (parent && recursive) {
+			const count = await this.model.find({ parent }).countDocuments();
+			if (count === 0) {
+				return [];
+			}
+			const parentId = new ObjectId(parent);
+			const post = await this.model.aggregate([
+				{
+					$graphLookup: {
+						from: 'posts',
+						startWith: '$_id',
+						connectFromField: '_id',
+						connectToField: 'parent',
+						as: 'children',
+						maxDepth: parseInt(process.env.MAX_COMMENT_DEPTH) - 1,
+						depthField: 'level'
+					}
+				},
+				{
+					$unwind: '$children'
+				},
+				{
+					$sort: { 'children.level': -1 }
+				},
+				{
+					$group: {
+						_id: '$_id',
+						children: { $push: '$children' }
+					}
+				},
+				{
+					$addFields: {
+						children: {
+							$reduce: {
+								input: '$children',
+								initialValue: {
+									currentLevel: -1,
+									currentLevelPosts: [],
+									previousLevelPosts: []
+								},
+								in: {
+									$let: {
+										vars: {
+											prev: {
+												$cond: [
+													{ $eq: ['$$value.currentLevel', '$$this.level'] },
+													'$$value.previousLevelPosts',
+													'$$value.currentLevelPosts'
+												]
+											},
+											current: {
+												$cond: [
+													{ $eq: ['$$value.currentLevel', '$$this.level'] },
+													'$$value.currentLevelPosts',
+													[]
+												]
+											}
+										},
+										in: {
+											currentLevel: '$$this.level',
+											previousLevelPosts: '$$prev',
+											currentLevelPosts: {
+												$concatArrays: [
+													'$$current',
+													[
+														{
+															$mergeObjects: [
+																'$$this',
+																{ children: { $filter: { input: '$$prev', as: 'e', cond: { $eq: ['$$e.parent', '$$this._id'] } } } }
+															]
+														}
+													]
+												]
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				},
+				{
+					$addFields: { children: '$children.currentLevelPosts' }
+				},
+				{
+					$match: {
+						_id: parentId
 					}
-					await this.getAllPosts(post._id, depth + 1, post);
 				}
-			}
+			]).exec();
+			const [{ children: posts }] = post;
+			return posts;
+		} else if (!recursive && parent) {
+			return this.model.find({ parent: parent }).lean().exec();
```

Comment:
You should use shorthand here.

```suggestion
			return this.model.find({ parent }).lean().exec();
```